### PR TITLE
Fix sponsored brands extra overriding the campaign section of sponsored brands

### PR DESCRIPTION
**Pull request by wbmgaye**
## Changes

- modified `Sponsored Brands - Extra`
- ran download schemas, generate-models, remove-unknown-types, and generate-client scripts

## Github Issue

Please link to the github issue.

## Testing

Please describe the tests that you ran to verify your changes. If possible please also provide steps to reproduce and screenshots of test evidence.

These tests should be live, **not** unit/integration tests.

## Reviewer Checklist

**Reviewer: please make sure that this has been filled out before approving**

-   [ ] Changes are appropriate for the ticket
-   [ ] Github Issue is linked to the PR and the PR title includes the ticket ID
-   [ ] Evidence of live testing has been provided by the PR author
-   [ ] Reviewer has done live verification/testing and evidence has been provided as a comment below

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts [hunk 1]
```diff
@@ -13,7 +29,151 @@ describe('SponsoredBrandsClient', () => {
...
+
+		it('should create campaign of type SBCreateCampaignWithKeywords', async () => {
+			const creativeAsset: SBCreative = {
+				brandLogoAssetID: 'AWz-C_2InEZrJWCZIR5a',
+				brandName: 'Sample Creative Brand Name',
+				headline: 'Sample Creative Headline',
+			};
+
+			const landingPage: SBLandingPage = {
+				asins: ['B018UQ5AMS', 'B07PC7MHQ8', 'B07C1XC3GF'],
```
Comment (josh-fisher) [inline tests/integration/apis/clients/sponsored-brands-client.spec.ts:60]: Not sure these asins should be left in here.

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts [hunk 2]
```diff
@@ -13,7 +29,151 @@ describe('SponsoredBrandsClient', () => {
...
+				name: 'SBCreateCampaignWithKeywords_1',
+				budget: 1,
+				budgetType: BudgetType.Daily,
+				adFormat: AdFormat.ProductCollection,
+				startDate: '20220114',
+				endDate: '20220114',
+				keywords: [{ keywordText: 'Test', matchType: MatchType.Exact, bid: 0 }],
+				negativeKeywords: [{ keywordText: 'Bad', matchType: NegativeMatchType.NegativeExact }],
+				creative: creativeAsset,
+				brandEntityId: 'ENTITY1233121',
```
Comment (josh-fisher) [inline tests/integration/apis/clients/sponsored-brands-client.spec.ts:73]: Is this a generic entitity?

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts [hunk 3]
```diff
@@ -13,7 +29,151 @@ describe('SponsoredBrandsClient', () => {
...
+	describe('#createCampaignsSb', () => {
+		// Test input gathered from this page -
+		// https://advertising.amazon.com/API/docs/en-us/bulksheets/sb/sb-examples/sb-create
+		// We need to override generated headers because Amazon docs are incorrect
+		// - returns 415 error with the generated headers
+		const options = { headers: { ['Content-Type']: 'application/json' } };
+
+		it('should create campaign of type SBCreateCampaignWithKeywords', async () => {
+			const creativeAsset: SBCreative = {
+				brandLogoAssetID: 'AWz-C_2InEZrJWCZIR5a',
```
Comment (josh-fisher) [inline tests/integration/apis/clients/sponsored-brands-client.spec.ts:54]: Is this a generic assetID?

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts [hunk 4]
```diff
@@ -13,7 +29,151 @@ describe('SponsoredBrandsClient', () => {
...
+
+				options
+			);
+
+			expect(result.data).toBeDefined();
+		});
+
+		it('should create campaign of type SBCreateCampaignWithExpressions', async () => {
+			const creativeAsset: SBCreative = {
+				brandLogoAssetID: 'AWz-C_2InEZrJWCZIR5a',
```
Comment (josh-fisher) [inline tests/integration/apis/clients/sponsored-brands-client.spec.ts:92]: Generic?

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts [hunk 5]
```diff
@@ -13,7 +29,151 @@ describe('SponsoredBrandsClient', () => {
...
+				brandName: 'Sample Creative Brand Name',
+				headline: 'Sample Creative Headline',
+			};
+
+			const landingPage: SBLandingPage = {
+				asins: ['B018UQ5AMS', 'B07PC7MHQ8', 'B07C1XC3GF'],
```
Comment (josh-fisher) [inline tests/integration/apis/clients/sponsored-brands-client.spec.ts:98]: Should these remain or be part of .env config?

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts [hunk 6]
```diff
@@ -13,7 +29,151 @@ describe('SponsoredBrandsClient', () => {
...
+				name: 'SBCreateCampaignWithExpressions_1',
+				budget: 1,
+				budgetType: BudgetType.Daily,
+				adFormat: AdFormat.ProductCollection,
+				startDate: '20220114',
+				endDate: '20220114',
+				targets: [{ expressions: [{ type: ProductPredicateType.AsinSameAs, value: 'TestValue' }], bid: 0 }],
+				negativeTargets: [{ expressions: [{ type: ProductPredicateType.AsinSameAs, value: 'NegTestValue' }] }],
+				creative: creativeAsset,
+				brandEntityId: 'ENTITY1233121',
```
Comment (josh-fisher) [inline tests/integration/apis/clients/sponsored-brands-client.spec.ts:111]: Generic or should move to .env config?

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts [hunk 7]
```diff
@@ -13,7 +29,151 @@ describe('SponsoredBrandsClient', () => {
...
+
+			const campaign: SBVideoCreateCampaignWithExpressions = {
+				name: 'SBVideoCreateCampaignWithExpressions_1',
+				budget: 1,
+				budgetType: BudgetType.Daily,
+				adFormat: AdFormat.Video,
+				creative: creativeAsset,
+				startDate: '20220114',
+				endDate: '20220114',
+				brandEntityId: 'ENTITY1233121',
```
Comment (josh-fisher) [inline tests/integration/apis/clients/sponsored-brands-client.spec.ts:138]: Same as other questions

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts [hunk 8]
```diff
@@ -13,7 +29,151 @@ describe('SponsoredBrandsClient', () => {
...
+
+			const campaign: SBVideoCreateCampaignWithKeywords = {
+				name: 'SBVideoCreateCampaignWithKeywords_1',
+				budget: 1,
+				budgetType: BudgetType.Daily,
+				adFormat: AdFormat.Video,
+				creative: creativeAsset,
+				startDate: '20220114',
+				endDate: '20220114',
+				brandEntityId: 'ENTITY1233121',
```
Comment (josh-fisher) [inline tests/integration/apis/clients/sponsored-brands-client.spec.ts:164]: Same as others

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts (hunk 1 above)
Comment (wbmgaye) [reply tests/integration/apis/clients/sponsored-brands-client.spec.ts:60]: These are generic values, I copied from the API docs. Nothing client specific.

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts (hunk 2 above)
Comment (wbmgaye) [reply tests/integration/apis/clients/sponsored-brands-client.spec.ts:73]: These are generic values, I copied from the API docs. Nothing client specific.

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts (hunk 3 above)
Comment (wbmgaye) [reply tests/integration/apis/clients/sponsored-brands-client.spec.ts:54]: These are generic values, I copied from the API docs. Nothing client specific.

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts (hunk 5 above)
Comment (wbmgaye) [reply tests/integration/apis/clients/sponsored-brands-client.spec.ts:98]: These are generic values, I copied from the API docs. Nothing client specific.

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts (hunk 7 above)
Comment (wbmgaye) [reply tests/integration/apis/clients/sponsored-brands-client.spec.ts:138]: These are generic values, I copied from the API docs. Nothing client specific.

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts (hunk 8 above)
Comment (wbmgaye) [reply tests/integration/apis/clients/sponsored-brands-client.spec.ts:164]: These are generic values, I copied from the API docs. Nothing client specific.

File: tests/integration/apis/clients/sponsored-brands-client.spec.ts (hunk 4 above)
Comment (wbmgaye) [reply tests/integration/apis/clients/sponsored-brands-client.spec.ts:92]: These are generic values, I copied from the API docs. Nothing client specific.

**Review by josh-fisher — APPROVED** [review]
LTGM - All tests work as expected. Approving

**Pull request merged**